File: workers/cartographer/testdata/sample.ts
// Sample TypeScript file for Cartographer AST extraction tests.

export class DataProcessor {
  constructor(private config: Config) {}

  process(record: unknown): unknown {
    return record;
  }
}

export interface Config {
  path: string;
}

// Bare (non-exported) interface.
interface RawRecord {
  id: number;
  data: unknown;
}

export function loadConfig(path: string): Config {
  return { path };
}

export const formatValue = (val: unknown): string => String(val);

const counter = 0;

File: workers/cartographer/testdata/sample.rb
# Sample Ruby file for Cartographer AST extraction tests.

module Utilities
  def self.format(val)
    val.to_s
  end
end

class DataProcessor
  def initialize(config)
    @config = config
  end

  def process(record)
    record
  end
end

class Config
  attr_reader :path

  def initialize(path)
    @path = path
  end
end

def load_config(path)
  Config.new(path)
end
